app/assets/javascripts/student/interventions_controller.js

```diff
@@ -0,0 +1,194 @@
+    bindListeners: function () {
+      // interventions
+      this.$el.on('click', '#open-intervention-form', this.onAddNewIntervention.bind(this));
+      this.$el.on('click', '#close-intervention-form', this.onCancelNewIntervention.bind(this));
+      this.$el.on('click', '.intervention-cell', this.onSelectedIntervention.bind(this));
+      this.$el.on('ajax:success', '.new-intervention-form', this.onNewInterventionSaveSucceeded.bind(this));
+      this.$el.on('ajax:error', '.new-intervention-form', this.onNewInterventionSaveFailed.bind(this));
+
+      // progress notes
+      this.$el.on('click', '.add-progress-note', this.onNewProgressNoteClicked.bind(this));
+      this.$el.on('click', '.cancel-progress-note', this.onCancelNewProgressNote.bind(this));
+      this.$el.on('ajax:success', '.new-progress-note-form', this.onNewProgressNoteSaveSucceeded.bind(this));
+      this.$el.on('ajax:error', '.new-progress-note-form', this.onNewProgressNoteSaveFailed.bind(this));
+    },
+
+    onNewProgressNoteClicked: function(e) {
+      this.isAddingProgressNote = true;
+      this.render();
+    },
+
+    onCancelNewProgressNote: function(e) {
+      this.isAddingProgressNote = false;
+      this.render();
+    },
+
+    onSelectedIntervention: function (e) {
+      var interventionId = $(e.currentTarget).data('id');
+      this.selectedInterventionId = interventionId;
+      this.isAddingProgressNote = false;
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onAddNewIntervention: function (e) {
+      this.selectedInterventionId = null;
+      this.isShowingNewIntervention = true;
+      this.render();
+    },
+
+    onCancelNewIntervention: function (e) {
+      this.selectedInterventionId = this.defaultSelectedIntervention();
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onNewInterventionSaveSucceeded: function (e, data, status, xhr) {
+      this.interventions = [data].concat(this.interventions);
+      this.selectedInterventionId = this.defaultSelectedIntervention();
+      this.isShowingNewIntervention = false;
+      this.render();
+    },
+
+    onNewProgressNoteSaveSucceeded: function(e, data, status, xhr) {
+      var intervention = this.getSelectedIntervention();
+      intervention.progress_notes = intervention.progress_notes.concat([data]);
+      this.isAddingProgressNote = false;
+      this.render();
+    },
+
+    onNewProgressNoteSaveFailed: function(e, xhr, status, error) {
+      var $errorsEl = $(e.currentTarget).parent().find('.alert.errors');
+      this.insertErrorMessages($errorsEl, xhr.responseJSON.errors);
+    },
+
+    onNewInterventionSaveFailed: function(e, xhr, status, error) {
+      var $errorsEl = $(e.currentTarget).parent().find('.alert.errors');
+      this.insertErrorMessages($errorsEl, xhr.responseJSON.errors);
+    },
+
+    // This inserts the errors into the form to preserve the DOM state on the form,
```

Comment:
This is a place where using React to do a full re-render would be helpful.  Since we're not keeping the form state anywhere but in the DOM, we can't just add the errors to the state and re-render.  So this inserts the error message into the DOM directly, which is cool and lets us keep using the good parts of the existing Rails forms.